### PR TITLE
Variable View UI Cleanup part 1

### PR DESCRIPTION
**Pull request by IanMatthewHuff**
For #4356 

-   [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [x] Title summarizes what is changing.
-   [ ] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [x] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

**Comment by IanMatthewHuff** [comment]
A bit of context here. This change does two things. When hosted in a view (instead of a document) the variable explorer now no longer has a close button and sizes it self appropriate to the given view area. 

Change is a bit funkier then might be expected due to how the variable view works. The grid that we used needed a hard code vertical size. So we had lots of old code (Tim worked on this) to handle resizing vertically with a dragger. It's all a bit non-clean / non reacty. Given that we're probably going to deprecate this view the goal was to fit this into the old context as cleanly as possible with the least disturbance to the existing variable code paths. Basics is that the variable view panel watches for resizes, then sets a viewHeight value in the state that is only used then in view mode, which is when the variable explorer is hosted in a view as opposed to a document. 

Old Code (close button, doesn't fill space, has a draggable divider at the bottom):
image

New Code:
image

File: src/datascience-ui/interactive-common/variableExplorer.tsx [hunk 1]
```diff
@@ -178,15 +180,57 @@ export class VariableExplorer extends React.Component<IVariableExplorerProps, IV
...
+        ){
+            return true;
+        }
+
         return false;
     }
 
     public render() {
+        // This control renders differently when hosted in a versus a document
+        if (this.props.viewMode) {
```
Comment (IanMatthewHuff) [inline src/datascience-ui/interactive-common/variableExplorer.tsx:201]: This is where we split the rendering. There were a number of differences, so I just used two different functions for the view mode versus document mode.

File: src/datascience-ui/interactive-common/variableExplorer.tsx [hunk 2]
```diff
@@ -234,6 +278,13 @@ export class VariableExplorer extends React.Component<IVariableExplorerProps, IV
     }
 
     private renderGrid() {
+        let newGridHeight: number | undefined;
+
+        // In in view mode, just use the viewHeight prop for calculating size
+        if(this.props.viewMode) {
```
Comment (IanMatthewHuff) [inline src/datascience-ui/interactive-common/variableExplorer.tsx:284]: Previously variableExplorer kept an internal state outside of redux for setting vertical height. For the new view mode we don't need this, so we can just look at the viewHeight prop for sizing.

File: src/datascience-ui/variable-view/variableViewPanel.tsx [hunk 3]
```diff
@@ -50,12 +59,27 @@ export class VariableViewPanel extends React.Component<IVariableViewPanelProps>
...
         return (
-            <div id="variable-view-main-panel" role="Main" style={dynamicFont}>
+            <div id="variable-view-main-panel" role="Main" ref={this.panelRef} style={dynamicFont}>
                 {this.renderVariablePanel(this.props.baseTheme)}
             </div>
-        ); // NOTE: Currently the OPEN, button just exists to mimic the toggling of the variable view, make it easier to test when working
+        );
     }
 
+    private windowResized = () => {
```
Comment (IanMatthewHuff) [inline src/datascience-ui/variable-view/variableViewPanel.tsx:68]: Same resize / timer code that we used in other spots reactSlickGrid / monacoEditor.

File: src/datascience-ui/history-react/interactivePanel.tsx [hunk 4]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (IanMatthewHuff) [inline src/datascience-ui/history-react/interactivePanel.tsx:1]: this seems to spuriously trigger all over in .tsx files. Should I look into it more? I just disabled in .tsx files that I used. Might be able to turn it off for all of them, but I don't know if that would hide legit errors.

Comment (rchiodo) [reply src/datascience-ui/history-react/interactivePanel.tsx:1]: unused vars should start with _. They do in the .ts files. So I think it's legit?

File: src/datascience-ui/interactive-common/variableExplorer.tsx (hunk 1 above)
Comment (rchiodo) [reply src/datascience-ui/interactive-common/variableExplorer.tsx:199]: I'm not sure which one is which? Is view mode inside of a webview?

**Review by rchiodo — APPROVED** [review]

File: src/datascience-ui/interactive-common/variableExplorer.tsx (hunk 1 above)
Comment (IanMatthewHuff) [reply src/datascience-ui/interactive-common/variableExplorer.tsx:199]: @rchiodo Yeah, in retrospect the name isn't the best, view is too general. I was thinking view versus document...maybe standaloneMode? That's really the crux of the issue here as the variable view is hosted all by itself outside of the document context.

Comment (IanMatthewHuff) [reply src/datascience-ui/interactive-common/variableExplorer.tsx:199]: variableWebviewViewMode seems descriptive, but too wordy.

Comment (rchiodo) [reply src/datascience-ui/interactive-common/variableExplorer.tsx:199]: standaloneMode gets my vote. That's clear to me at least.

**Review by DavidKutu — APPROVED** [review]
